From 8567d62c58923b1342a580d5c30ac736e1a32aa2 Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Tue, 30 Apr 2013 16:09:33 +0300
Subject: [PORT FROM MAIN] Telephony: Clear CallForwarding and Message waiting Indications

BZ: 64610

When SIM is removed the SIMRecords object is disposed, but
the relevant events to clear the Callforwarding and message
waiting icons are not send.

This patch clears the call forwarding and message waiting
indicators by clearing the internal member data and sending
out notifications for the change in data.

Change-Id: I535b0aa1ba740cf44dc1442a9ddb3e98cf7e8912
Orig-Change-Id: I9a123c8fbeccf2c6a11d1d60a58eeb9898f25cdb
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/uicc/SIMRecords.java        |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/uicc/SIMRecords.java b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
index 9be2da1..72b4d56 100644
--- a/src/java/com/android/internal/telephony/uicc/SIMRecords.java
+++ b/src/java/com/android/internal/telephony/uicc/SIMRecords.java
@@ -233,6 +233,10 @@ public class SIMRecords extends IccRecords {
         mSpn = null;
         mEfMWIS = null;
         mEfCPHS_MWI = null;
+        setVoiceMessageWaiting(1, 0);
+        mEfCfis = null;
+        mEfCff = null;
+        setVoiceCallForwardingFlag(1, false, null);
         mSpdiNetworks = null;
         mPnnHomeName = null;
         mGid1 = null;
-- 
1.7.4.1

